File: go/password_validation.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	fmt.Print("What is the password? ")
	scanner.Scan()
	password := scanner.Text()

	if password == "abc$123" {
		fmt.Println("Welcome!")
	} else {
		fmt.Println("I don't know you.")
	}
}

File: c/password_validation.c
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(void) {
  char password[20];

  printf("What is the password? ");
  scanf("%s", password);

  if (!strcmp(password, "abc$123")) {
    puts("Welcome!");
  } else {
    puts("I dont' know you.");
  }

  return EXIT_SUCCESS;
}
